### PR TITLE
kafka spout patching

### PR DESCRIPTION
**Pull request by nlu90**
Patching heron for it work with kafka-spout.

### CODE REVIEW

**Comment by mycFelix** [comment]
Great!!!

File: heron/storm/src/java/org/apache/storm/Config.java [hunk 1]
```diff
@@ -260,6 +260,53 @@
...
+
+  /**
+   * The port Storm will use to connect to each of the ZooKeeper servers.
+   */
+  public static final String STORM_ZOOKEEPER_PORT = "storm.zookeeper.port";
+
+  /**
+   * The root directory in ZooKeeper for metadata about TransactionalSpouts.
+   */
+  public static final String TRANSACTIONAL_ZOOKEEPER_ROOT="transactional.zookeeper.root";
```
Comment (objmagic) [inline heron/storm/src/java/org/apache/storm/Config.java]: umm, to keep uniformity, should we surround assignment operator using whitespace?

**Comment by objmagic** [comment]
👍

**Comment by kramasamy** [comment]
@nlu90 - This patch will suffice for running Storm kafka spout. Or we need this for heron-kafka spout as well?

File: heron/storm/src/java/org/apache/storm/Config.java (hunk 1 above)
Comment (nlu90) [inline heron/storm/src/java/org/apache/storm/Config.java]: good find

**Comment by nlu90** [comment]
@kramasamy we need this for heron-kafka spout as well

**Comment by kramasamy** [comment]
@nlu90 - instead of introducing another set of configuration for the spout, I was wondering, can we think about making it more simpler. Looking at the config - kafka spout - seems to adding a lot of config and hard coded as well. We want to move away from hard coding the config into the topology. We can introduce the notion of config per component - in a cleaner fashion.

**Comment by nlu90** [comment]
@kramasamy If we want to introduce the notion of config per component, what are the changes needed?

**Comment by kramasamy** [comment]
ship it 👍

**Pull request merged**